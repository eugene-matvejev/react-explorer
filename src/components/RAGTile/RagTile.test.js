import React from 'react';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import RagTile from './RagTile';
import {TileHeader, TileWrapper} from './RagTileStyling';

configure({ adapter: new Adapter() });

describe('RagTile', () => {

  const props = {
    id: 1,
    status : "red",
    name : "Health Check 1",
    parent: null,
    onClick: jest.fn(),
  }

  const propsTileWrapper = {
    ...{ key : props.id, tileColour: props.status, onClick : props.onClick}
  }

  let mountedRagTile;

  beforeEach(() => {
    mountedRagTile = shallow(<RagTile {...props} />);
  });

  describe('renders', () => {
    
    it('with some props', () => {
      expect(mountedRagTile).toMatchSnapshot();
    });

    it('a TileWrapper Component with the correct props', () => {
      const tileWrapper = mountedRagTile.find(TileWrapper);
      expect(tileWrapper.length).toBe(1);
      expect(tileWrapper.props().tileColour).toEqual('red');
      expect(tileWrapper.props().onClick).toEqual(expect.any(Function));
    });

    it('a TileHeader Component', () => {
      const tileHeader = mountedRagTile.find(TileHeader);
      expect(tileHeader.length).toBe(1);   
    });


  });
});

