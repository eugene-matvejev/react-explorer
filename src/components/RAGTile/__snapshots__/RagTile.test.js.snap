// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RagTile renders with some props 1`] = `
<styled.section
  onClick={[Function]}
  tileColour="red"
>
  <styled.h1>
     
    Health Check 1
     
  </styled.h1>
</styled.section>
`;
